automates/text_reading/src/test/scala/org/clulab/aske/automates/text/TestFunctions.scala

```diff
@@ -96,31 +140,189 @@ class TestFunctions extends ExtractionTest {
+    val desired = Seq(
+      "Hourly canopy photosynthesis on a land area basis" -> Seq("sunlit and shaded leaf contributions"),
+      "Hourly canopy photosynthesis on a land area basis" -> Seq("sunlit and shaded leaf photosynthetic rates", "their respective LAIs")
+    )
+    val mentions = extractMentions(t6f)
+    testFunctionEvent(mentions, desired)
+  }
+
+  val t7f = "Maintenance respiration depends on temperature as well as gross photosynthesis and total crop mass minus protein and oil in the seed."
+  failingTest should s"find functions from t7f: ${t7f}" taggedAs(Somebody) in {
+    val desired = Seq(
+      "Maintenance respiration" -> Seq("temperature", "gross photosynthesis", "total crop mass minus protein", "oil in the seed")
+    )
+    val mentions = extractMentions(t7f)
+    testFunctionEvent(mentions, desired)
+  }
+
+  val t8f = "Daily growth rate is modified by temperature and assimilate availability."
+  failingTest should s"find functions from t8f: ${t8f}" taggedAs(Somebody) in {
+    val desired = Seq(
+      "Daily growth rate" -> Seq("temperature", "assimilate availability") //fixme: "assimilate availability" is not captured as an input (bad parsing)
+    )
+    val mentions = extractMentions(t8f)
+    testFunctionEvent(mentions, desired)
+  }
+
+  // model for predicting evaporation from a row crop with incomplete cover
+  val t1g = "The crop evaporation rate is calculated by adding the soil surface and plant surface components (each of these requiring daily numbers for the leaf area index), the potential evaporation, the rainfall, and the net radiation above the canopy."
+  failingTest should s"find functions from t1g: ${t1g}" taggedAs(Somebody) in {
+    val desired = Seq(
+      "crop evaporation rate" -> Seq("soil surface components", "plant surface components", "the potential evaporation", "the rainfall", "the net radiation above the canopy") // fixme: This example has a conjunction issue.
+    )
+    val mentions = extractMentions(t1g)
+    testFunctionEvent(mentions, desired)
+  }
+
+  val t2g = "The evaporation from the soil surface Es is calculated in two stages: (1) the constant rate stage in which Es is limited only by the supply of energy to the surface and (2) the falling rate stage in which water movement to the evaporating sites near the surface is controlled by the hydraulic properties of the soil."
+  passingTest should s"find functions from t2g: ${t2g}" taggedAs(Somebody) in {
+    val desired = Seq(
+      "Es" -> Seq("supply of energy to the surface"), // fixme: one additional, unwanted concept is captured as an input due to bad parsing.
+      "water movement to the evaporating sites near the surface" -> Seq("hydraulic properties of the soil")
+    )
+    val mentions = extractMentions(t2g)
+    testFunctionEvent(mentions, desired)
+  }
+
+  val t3g = "The model was used to obtain the total evaporation rate E = Es + Ep of a developing grain sorghum (Sorghum bicolor L.) canopy in central Texas."
```

Comment:
Do we want to extract the descriptions in functions? I can see it going both ways (e.g., to disambiguate E from other Es vs. "well, why extract descriptions if we extract them elsewhere?")

Reply:
We will keep this as it is now.